src/data-fetcher-loop/signed-data-state.ts

```diff
@@ -1,100 +1,80 @@
 
-  if (!goVerify.success) {
-    logger.error(`Signature verification failed.`, {
-      signature,
-      timestamp,
-      encodedValue,
-    });
+  // Check that the signed data is fresher than the one stored in state.
+  const timestampMs = Number(timestamp) * 1000;
+  const storedValue = getState().signedDatas[beaconId];
+  if (storedValue && Number(storedValue.timestamp) * 1000 >= timestampMs) {
+    logger.debug('Skipping state update. The signed data value is not fresher than the stored value.');
     return false;
   }
 
-  return true;
-};
-
-const verifyTimestamp = (timestamp: number) => {
-  const timestampMs = timestamp * 1000;
-
-  if (timestampMs > Date.now() + 60 * 60 * 1000) {
+  // Verify the timestamp of the signed data.
+  const nowMs = Date.now();
+  if (timestampMs > nowMs + 60 * 60 * 1000) {
     logger.error(`Refusing to store sample as timestamp is more than one hour in the future.`, {
-      systemDateNow: new Date().toLocaleDateString(),
-      signedDataDate: new Date(timestampMs).toLocaleDateString(),
+      airnode,
+      templateId,
+      timestampMs,
+      nowMs,
     });
     return false;
   }
-
-  if (timestampMs > Date.now()) {
+  if (timestampMs > nowMs) {
     logger.warn(`Sample is in the future, but by less than an hour, therefore storing anyway.`, {
-      systemDateNow: new Date().toLocaleDateString(),
-      signedDataDate: new Date(timestampMs).toLocaleDateString(),
+      airnode,
+      templateId,
+      timestampMs,
+      nowMs,
     });
   }
 
   return true;
 };
 
-export const verifySignedDataIntegrity = (signedData: SignedData) => {
-  // TODO: Temporarily disable signed data verification.
-  return verifyTimestamp(Number.parseInt(signedData.timestamp, 10)) /* && verifySignedData(signedData) */;
-};
+export const saveSignedData = async (signedDataBatch: SignedDataRecordEntry[]) => {
+  // Filter out signed data with invalid timestamps or we already have a fresher signed data stored in state.
+  signedDataBatch = signedDataBatch.filter((signedDataEntry) => verifyTimestamp(signedDataEntry));
+  if (signedDataBatch.length === 0) return;
 
-export const saveSignedData = (signedData: SignedData) => {
-  const { airnode, templateId, timestamp } = signedData;
-
-  // Make sure we run the verification checks with enough context.
-  logger.runWithContext({ airnode, templateId }, () => {
-    if (!verifySignedDataIntegrity(signedData)) {
-      return;
-    }
-
-    const state = getState();
-
-    const dataFeedId = deriveBeaconId(airnode, templateId) as Hex;
-
-    const existingValue = state.signedDatas[dataFeedId];
-    if (existingValue && existingValue.timestamp >= timestamp) {
-      logger.debug('Skipping state update. The signed data value is not fresher than the stored value.');
-      return;
+  const verifier = await getVerifier();
+  const verificationResult = await verifier.verifySignedData(signedDataBatch);
+  if (verificationResult !== true) {
```

Comment:
We skipping whole batch if one of them is defective. However, it looks okay for me, we don't expect any faulty signed data.

Reply:
Yeah, that's questionable part, but it felt natural to me because we are doing the same in the Signed API https://github.com/api3dao/signed-api/blob/main/packages/signed-api/README.md#api

I'll leave a comment about it though.